tools/common/OSPlatformAttributeExtensions.cs

```diff
@@ -0,0 +1,119 @@
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.Versioning;
+
+using Xamarin.Utils;
+
+#nullable enable
+
+namespace Xamarin.Utils {
+	internal static class OSPlatformAttributeExtensions {
+#if NET
+		public static bool TryParse (string? platformName, [NotNullWhen (true)] out string? platform, out Version? version)
```

Comment:
This attribute can be stubbed out with an internal version of the nullability attributes in the assembly:
https://source.dot.net/#Microsoft.Extensions.Logging.Console/NullableAttributes.cs,c0596421bc26d09e

Reply:
It also works for codeanalysis as a result when targeting .netframework